manual_flagging/initial_flagging.py

```diff
@@ -121,14 +130,167 @@
+            key_condo = f"df_condo_og_mansueto"
+            dfs_to_feature_creation[key_res] = df_res_og_mansueto
+            dfs_to_feature_creation[key_condo] = df_condo_og_mansueto
+            break
+
+# check names
+for key in dfs_to_feature_creation:
+    print(key)
+
+# - - - - - - -
+# make features
+# - - - - - - -
+
+
+def create_bins_and_labels(input_list):
+    # Initialize bins with 0 and float("inf")
+    bins = [0] + input_list + [float("inf")]
+
+    # Generate labels based on bins
+    labels = []
+    for i in range(len(bins) - 1):
+        if i == 0:
+            labels.append(f"below_{bins[i+1]}")
+        elif i == len(bins) - 2:
+            labels.append(f"above_{bins[i]}")
+        else:
+            labels.append(f"{bins[i]}_to_{bins[i+1]}")
+
+    return bins, labels
+
+
+# Iterate through loop and correctly create columns that are used for statistical grouping
+dfs_to_rolling_window = {}
+for df_name, df in dfs_to_feature_creation.items():
+    # Make a copy of the data frame to edit
+    df_copy = df.copy()
+
+    if "current" in df_name:
+        print("current")
+
+        if "res_single_family" in df_name:
+            # Bin sf
+            char_bldg_sf_bins, char_bldg_sf_labels = create_bins_and_labels(
+                inputs["stat_groups"]["current"]["res_single_family"][
+                    "sf_bin_specification"
+                ]
+            )
+            df_copy["char_bldg_sf_bin"] = pd.cut(
+                df_copy["char_bldg_sf"],
+                bins=char_bldg_sf_bins,
+                labels=char_bldg_sf_labels,
+            )
+            # Define bins for building age
+            bldg_age_bins, bldg_age_labels = create_bins_and_labels(
+                inputs["stat_groups"]["current"]["res_single_family"][
+                    "age_bin_specification"
+                ]
+            )
+            df_copy["bldg_age_bin"] = pd.cut(
+                df_copy["bldg_age"],
+                bins=char_bldg_sf_bins,
+                labels=char_bldg_sf_labels,
+            )
+
+        if "res_multi_family" in df_name:
+            # Define bins for building age
+            bldg_age_bins, bldg_age_labels = create_bins_and_labels(
+                inputs["stat_groups"]["current"]["res_multi_family"][
+                    "age_bin_specification"
+                ]
+            )
+            df_copy["bldg_age_bin"] = pd.cut(
+                df_copy["bldg_age"],
+                bins=char_bldg_sf_bins,
+                labels=char_bldg_sf_labels,
+            )
+            pass
+
+        if "condos" in df_name:
+            pass
+
+    elif "og_mansueto" in df_name:
+        print("og_mansueto")
+        if "res" in df_name:
+            # Process residential data here on df_copy
+            pass
+
+        if "condos" in df_name:
+            pass
+
+    # Add the edited or unedited dataframe to the new dictionary
+    dfs_to_rolling_window[df_name] = df_copy
+
+
+# Make rolling window
+#
+#
```

Comment:
So far I have just implemented filters and feature creation. My next step would be to move to rolling_window creation and the actual flagging.